### PR TITLE
Missile Sanity Checking: I made it explode better?

### LINKED ISSUE #70809: Missile ammo box subtypes are backwards
## Reproduction:

Confusingly, `/obj/item/mecha_ammo/missiles_he`, also known as "anti armor missiles", which HAVE `HE` IN THEIR TYPE, are actually `MECHA_AMMO_MISSILE_AP`, which the BRM-6 uses. Despite this, the description of it implies the SRM-8 uses it. 

CONVERSELY, `/obj/item/mecha_ammo/missiles_br`, called "breaching missiles", are ACTUALLY `MECHA_AMMO_MISSILE_HE`, which the SRM-8 uses. Despite this, the description of it implies the BRM-6 uses it. 

What does this mean?

Well, the nukie "Mauler Ammunition Bag" contains three cases of missiles that the Mauler cannot use. They are `missiles_he`, which are anti-armor / AP, which the SRM-8 (the Mauler's missile rack) cannot use

Also, on-station roboticists can only print the incorrect ammo type for the only missile type they can create. Roboticists can create BRM-6s, but can only print `missiles_he`, which are "breaching missiles", which the BRM-6 cannot use.

How this happened I'm not sure but it's incredibly silly.

### PR DESCRIPTION
**Pull request by necromanceranne**
## About The Pull Request

Fixes https://github.com/tgstation/tgstation/issues/70809

While doing so, makes it much clearer which missile type goes into which missile rack.

Additional changes include:

### **Renames the BRM-6 to the PEP-6**
It was a nonsense name before, but now it stands for Precise Explosive Projectile. Named in the fashion of a famous weapon everyone loves to use during stealth missions.

### **Refactors rockets and missiles so that it is no longer using an upgraded missile type for nukies as a parent type** 
Why this was done is beyond me, but basically resolves some weirdness with subtyping and unnecessary var editing

### **Makes the PEP missile a subtype of the normal rocket and not its own snowflake rocket**
Since the other rockets actually have very useful code for what the PEP advertises that it does, I made it a subtype of other rockets and with its own explosion rules. 

**However**, this isn't with total parity. It now does a boatload of damage to borgs, and explodes a considerable amount of additional dense objects. But I'm honestly not sure why it couldn't do this to begin with since it is supposed to be an anti-materiel missile. So why were airlocks and mechs valid, but lockers, borgs and the janicart not? 

If maints would rather this _not_ happen, I can change it, but boy is it stupid that it works this way, and I have a pretty easy way of showing how it is inconsistent and unintuitive.

For context, it could explode: ~~doors~~, airlocks, windows, walls, grilles, mechs.

That's it. You shoot one of the many dense computers or lockers littering the station, and you're shit out of luck. But you hit a flimsy metal grille, it detonates. You can blow up mechs, but apparently cyborgs aren't valid. 

## Why It's Good For The Game

Going over this stuff, I really quickly realized how bizarre rocket code is, so here I am fixing it and making it work better.

## Changelog

:cl:
fix: Nukies and roboticists access the correct missile types again.
refactor: Cleans up the missile subtypes.
balance: Renames the BRM-6 to the PEP-6, and makes it explode...a lot of the stuff that it claims it should explode.
/:cl:

### CODE REVIEW

**Comment by necromanceranne** [comment]
Slight amendment: BRMs could destroy airlocks, but mineral doors were not included. Yes, even if made from metal materials or uranium or diamond.

Gosh I love consistency.

File: code/modules/projectiles/projectile/special/rocket.dm [hunk 1]
```diff
@@ -10,21 +10,20 @@
 	explosion(target, devastation_range = -1, light_impact_range = 2, explosion_cause = src)
 	return BULLET_ACT_HIT
 
-/// PM9 HEDP rocket
+/// PM9 standard HE rocket
 /obj/projectile/bullet/a84mm
```
Comment (Fikou) [inline code/modules/projectiles/projectile/special/rocket.dm:14]: can you rename this to rocket? annoying to var edit

File: code/modules/projectiles/projectile/special/rocket.dm [hunk 2]
```diff
@@ -39,67 +38,66 @@
...
-/obj/projectile/bullet/a84mm/he/do_boom(atom/target, blocked=0)
 	if(!isliving(target)) //if the target isn't alive, so is a wall or something
 		explosion(target, heavy_impact_range = 1, light_impact_range = 2, flame_range = 3, flash_range = 4, explosion_cause = src)
 	else
 		explosion(target, light_impact_range = 2, flame_range = 3, flash_range = 4,  explosion_cause = src)
 
-/// PM9 weak rocket
+/// PM9 HEAP rocket - the anti-anything missile you always craved.
+/obj/projectile/bullet/a84mm/heap
+	name ="\improper HEAP rocket"
```
Comment (Fikou) [inline code/modules/projectiles/projectile/special/rocket.dm:50]:
```suggestion
	name = "\improper HEAP rocket"
```

File: code/modules/projectiles/projectile/special/rocket.dm [hunk 3]
```diff
@@ -10,21 +10,20 @@
...
 /obj/projectile/bullet/a84mm
-	name ="\improper HEDP rocket"
-	desc = "USE A WEEL GUN"
-	icon_state= "84mm-hedp"
-	damage = 80
-	armour_penetration = 100
-	dismemberment = 100
-	embedding = null
-	shrapnel_type = null
+	name ="\improper HE rocket"
```
Comment (Fikou) [inline code/modules/projectiles/projectile/special/rocket.dm:15]:
```suggestion
	name = "\improper HE rocket"
```

File: code/modules/projectiles/projectile/special/rocket.dm [hunk 4]
```diff
@@ -39,67 +38,66 @@
...
+	dismemberment = 100
+	anti_armour_damage = 200
+
+/obj/projectile/bullet/a84mm/heap/do_boom(atom/target, blocked=0)
+	explosion(target, devastation_range = -1, heavy_impact_range = 1, light_impact_range = 3, flame_range = 4, flash_range = 1, adminlog = FALSE)
+
+/// PM9 weak rocket - just kind of a failure
 /obj/projectile/bullet/a84mm/weak
-	name ="low-yield HE missile"
+	name ="low-yield rocket"
```
Comment (Fikou) [inline code/modules/projectiles/projectile/special/rocket.dm:63]:
```suggestion
	name = "low-yield rocket"
```

File: code/modules/projectiles/projectile/special/rocket.dm [hunk 5]
```diff
@@ -39,67 +38,66 @@
...
 
-/// Mech BRM-6 missile
-/obj/projectile/bullet/a84mm_br
-	name ="\improper HE missile"
-	desc = "Boom."
-	icon_state = "missile"
+/// SRM-8 Missile - Used by the SRM-8 Exosuit missile rack.
+// Employed by Nuclear Operatives Maulers and Nanotrasen Marauders and Seraphs to kill everything and anyone.
+/obj/projectile/bullet/a84mm/srm
+	name ="\improper short range missile"
```
Comment (Fikou) [inline code/modules/projectiles/projectile/special/rocket.dm:76]:
```suggestion
	name = "\improper short range missile"
```

File: code/modules/projectiles/projectile/special/rocket.dm [hunk 6]
```diff
@@ -39,67 +38,66 @@
...
+	desc = "Today's not your day, pal."
+
+/// PEP-6 Missile - Used by the PEP-6 Exosuit missile rack.
+// Employed by Roboticists out of spite to put down enemy hereteks, mechanized nuclear operatives, the janitor's hot rod, the clown's 'taxi service', uppity borgs, vengeful ais, doors they don't like, the escape shuttle's hull, and more!
+/obj/projectile/bullet/a84mm/pep
+	name ="\improper precise explosive missile"
```
Comment (Fikou) [inline code/modules/projectiles/projectile/special/rocket.dm:82]:
```suggestion
	name = "\improper precise explosive missile"
```

File: code/modules/projectiles/projectile/special/rocket.dm [hunk 7]
```diff
@@ -39,67 +38,66 @@
...
 	else
 		explosion(target, light_impact_range = 2, flame_range = 3, flash_range = 4, explosion_cause = src)
 
-/// Mech BRM-6 missile
-/obj/projectile/bullet/a84mm_br
-	name ="\improper HE missile"
-	desc = "Boom."
-	icon_state = "missile"
+/// SRM-8 Missile - Used by the SRM-8 Exosuit missile rack.
+// Employed by Nuclear Operatives Maulers and Nanotrasen Marauders and Seraphs to kill everything and anyone.
```
Comment (Fikou) [inline code/modules/projectiles/projectile/special/rocket.dm:74]: im not sure using autodoc like that works, can you just do multiline version of autodoc?

**Review by Fikou — CHANGES_REQUESTED** [review]

File: code/modules/projectiles/projectile/special/rocket.dm (hunk 1 above)
Comment (necromanceranne) [reply code/modules/projectiles/projectile/special/rocket.dm:14]: Just the path? Sure thing.

File: code/modules/projectiles/projectile/special/rocket.dm (hunk 7 above)
Comment (necromanceranne) [reply code/modules/projectiles/projectile/special/rocket.dm:74]: Ye

**Comment by necromanceranne** [comment]
Thanks for reminding me about the icons.

I'm aware of the battletech association, but I know literally nothing about battletech, so I wasn't confident in how to keep that going. If there is a good equivalency, I'll totally change it to that, no problem. But just reading up on the suggested names, it seems to imply long range ballistic missiles rather than penetrative precision missiles. Maybe that's not really appropriate either? Anyone who knows a lot about Battletech, feel free to chip in here.

File: code/modules/projectiles/projectile/special/rocket.dm [hunk 8]
```diff
@@ -39,67 +38,77 @@
...
-	/obj/structure/grille
-	)
+	anti_armour_damage = 80 //Doesn't (probably) kill borgs in one shot, but it will hurt
+	random_crits_enabled = FALSE //yeah, no
+
+/obj/projectile/bullet/rocket/pep/do_boom(atom/target, blocked=0)
+	if(issilicon(target)) //if the target is a borg, just give them one of these to make it loud, most of the damage is in the projectile itself
+		explosion(target, light_impact_range = 1, flash_range = 2, explosion_cause = src)
+		return
+	if(isstructure(target) || isvehicle (target) || isclosedturf (target) || ismachinery (target)) //if the target is a structure, machine, vehicle or closed turf like a wall, explode that shit
```
Comment (Fikou) [inline code/modules/projectiles/projectile/special/rocket.dm:103]: make a weaker explosion or something if its not dense, light fixtures or something probably shouldnt cause a heavy impact

**Comment by Fikou** [comment]
let's see how this goes

**Pull request merged**